siliconcompiler/tools/vpr/place.py

```diff
@@ -38,7 +40,38 @@ def pre_process(chip):
     step = chip.get('arg', 'step')
     index = chip.get('arg', 'index')
 
-    if not chip.valid('input', 'constraint', 'pins', default_valid=True):
+    part_name = chip.get('fpga', 'partname')
+
+    # If the user explicitly provides an XML constraints file, give that
+    # priority over other constraints input types:
+    if 'pins' in chip.getkeys('input', 'constraint'):
+        constraint_file = vpr.find_single_file(chip, 'input', 'constraint', 'pins',
+                                               step=step, index=index,
+                                               file_not_found_msg="VPR constraints file not found")
+
+        if (constraint_file is not None):
+            shutil.copy(constraint_file, vpr.auto_constraints())
```

Comment:
I'm going to approve, but this is bad practice. If the file doesn't need to be moved around, we should always refer to the source material.